loki/transform/transform_inline.py

```diff
@@ -239,7 +239,12 @@ def _map_unbound_dims(var, val):
         {v: v.clone(name=f'{member.name}_{v.name}') for v in duplicate_locals}
     )
     member.spec = shadow_mapper.visit(member.spec)
-    member.body = shadow_mapper.visit(member.body)
+
+    var_map = {}
+    for v in FindVariables(unique=False).visit(member.body):
+        if v.name in [dl.name for dl in duplicate_locals]:
+            var_map[v] = v.clone(name=f'{member.name}_{v.name}')
```

Comment:
The reason that you need to do that list comprehension is that the (string) comparison takes into account dimension expressions of variables. If you want to make it slightly less error prone (because you would likely run into case-sensitivity issues with the current implementation): 

```suggestion
    duplicate_locals_names = {dl.name.lower() for dl in duplicate_locals}
    for v in FindVariables(unique=False).visit(member.body):
        if v.name.lower() in duplicate_locals_names:
            var_map[v] = v.clone(name=f'{member.name}_{v.name}')
```